server/src/test/java/org/elasticsearch/common/xcontent/support/XContentHelperTests.java

```diff
@@ -105,4 +106,104 @@ public void testToXContent() throws IOException {
             }
         }
     }
+
+    public void testChildBytes() throws IOException {
+
+        for (XContentType xContentType : XContentType.values()) {
+
+            // TODO how to deal with embedded objects?
```

Comment:
Is this something we need to think about in this PR?

Reply:
I don't think it is if we're using `copyCurrentStructure`. It'd be nice though.

Reply:
It looks as though we do use this in a couple of places, so I'll update the PR accordingly; it gets generated by `XContentBuilder.field(String, byte[])`